pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -132,9 +135,66 @@ func (r *Reconciler) create() error {
 		}
 		return fmt.Errorf("failed to create instance via compute service: %v", err)
 	}
+	// Create succeeded, add to target pool if necessary
+	if r.providerSpec.TargetPools != nil && len(r.providerSpec.TargetPools) > 0 {
+		instanceSelfLink := getInstanceSelfLink(r.projectID, zone, instance.Name)
+		for _, pool := range r.providerSpec.TargetPools {
+			err := r.addInstanceToTargetPool(instanceSelfLink, pool)
+			if err != nil {
+				return fmt.Errorf("failed to add instance %v to target pool %v: %v", instance.Name, pool, err)
+			}
+		}
+	}
 	return r.reconcileMachineWithCloudState(nil)
 }
 
+func (r *Reconciler) addInstanceToTargetPool(instanceLink string, pool string) error {
+	// Get target pool
+	tp, err := r.computeService.TargetPoolsGet(r.projectID, r.providerSpec.Region, pool)
+	if err != nil {
+			return err
+	}
+
+	// Check if instance is in target pool already
+	if tp.Instances != nil {
+		for _, link := range tp.Instances {
+			if instanceLink == link {
+				// Already in the TargetPool, do nothing.
+				return nil
+			}
+		}
+	}
+	// if not, add instance to target pool
+	_, err = r.computeService.TargetPoolsAddInstance(r.projectID, r.providerSpec.Region, pool, instanceLink)
+	// Probably safe to disregard the returned operation; it either worked or it didn't.
+	return err
+}
+
+// This function is probably overkill.  Probably should just use fmt.Sprintf
+func getInstanceSelfLink(project, zone, name string) string {
+	instanceInfo := struct {
+		Project string
+		Zone    string
+		Name    string
+	}{
+		Project: project,
+		Zone:    zone,
+		Name:    name,
+	}
+	tmpl, err := template.New("test").Parse(instanceLinkTString)
```

Comment:
Why isn't this just sprintf? That would get rid of the panic too.

Reply:
yeah, I didn't realize it was going to be so verbose to write it this way, but I did, lol.  I already planned on replacing it.